File: app_backup_refactored/src/main/java/com/chopcode/rutago/app/activities/common/ForgotPasswordActivity.java
package com.chopcode.rutago.app.activities.common;

import android.os.Bundle;
import android.util.Log;
import android.util.Patterns;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.chopcode.rutago.app.R;
import com.chopcode.rutago.app.viewmodels.common.ForgotPasswordViewModel;
import com.google.android.material.textfield.TextInputEditText;
import com.google.android.material.textfield.TextInputLayout;
import com.google.android.material.appbar.MaterialToolbar;

/**
 * 🔑 Forgot Password Activity
 */
public class ForgotPasswordActivity extends AppCompatActivity {

    private static final String TAG = "ForgotPasswordActivity";

    private TextInputEditText etEmail;
    private TextInputLayout tilEmail;
    private Button btnRecover, btnUnderstood;
    private MaterialToolbar topAppBar;
    private ProgressBar progressBar;
    private View formCardView, layoutSuccess;
    private TextView tvSubtitle;

    private ForgotPasswordViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d(TAG, "🚀 onCreate - Starting recovery flow");
        setContentView(R.layout.activity_recuperar_contrasena);

        viewModel = new ViewModelProvider(this).get(ForgotPasswordViewModel.class);

        initViews();
        setupToolbar();
        setupObservers();
        setupListeners();

        String prefilledEmail = getIntent().getStringExtra("email");
        if (prefilledEmail != null && !prefilledEmail.isEmpty()) {
            etEmail.setText(prefilledEmail);
        }
    }

    private void initViews() {
        etEmail = findViewById(R.id.et_email);
        tilEmail = findViewById(R.id.til_email);
        btnRecover = findViewById(R.id.btn_recuperar);
        topAppBar = findViewById(R.id.topAppBar);
        progressBar = findViewById(R.id.pb_loading);
        formCardView = findViewById(R.id.formCardView);
        layoutSuccess = findViewById(R.id.layout_success);
        tvSubtitle = findViewById(R.id.tv_subtitle);
        btnUnderstood = findViewById(R.id.btn_entendido);
    }

    private void setupToolbar() {
        if (topAppBar != null) {
            setSupportActionBar(topAppBar);
            topAppBar.setNavigationOnClickListener(v -> onBackPressed());
        }
    }

    private void setupObservers() {
        viewModel.getSendSuccess().observe(this, success -> { if (success) showSuccessState(); });
        viewModel.getSendError().observe(this, msg -> { btnRecover.setEnabled(true); Toast.makeText(this, "Error: " + msg, Toast.LENGTH_LONG).show(); });
        viewModel.getIsLoading().observe(this, loading -> { progressBar.setVisibility(loading ? View.VISIBLE : View.GONE); btnRecover.setEnabled(!loading); });
    }

    private void setupListeners() {
        btnRecover.setOnClickListener(v -> {
            String email = etEmail.getText().toString().trim();
            if (validateEmail(email)) viewModel.sendResetEmail(email);
        });
        btnUnderstood.setOnClickListener(v -> finish());
    }

    private boolean validateEmail(String email) {
        if (email.isEmpty()) { tilEmail.setError(getString(R.string.correo_obligatorio)); return false; }
        if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) { tilEmail.setError(getString(R.string.correo_valido)); return false; }
        tilEmail.setError(null);
        return true;
    }

    private void showSuccessState() {
        formCardView.setVisibility(View.GONE);
        tvSubtitle.setVisibility(View.GONE);
        if (getSupportActionBar() != null) getSupportActionBar().setTitle(getString(R.string.todo_listo));
        layoutSuccess.setVisibility(View.VISIBLE);
        layoutSuccess.setAlpha(0f);
        layoutSuccess.animate().alpha(1f).setDuration(600).start();
    }
}

File: app/src/main/java/com/chopcode/rutago/app/activities/common/ForgotPasswordActivity.kt
package com.chopcode.rutago.app.activities.common

import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.lifecycle.viewmodel.compose.viewModel
import com.chopcode.rutago.app.ui.screens.auth.ForgotPasswordScreen
import com.chopcode.rutago.app.ui.theme.RutaGoTheme
import com.chopcode.rutago.app.ui.viewmodels.auth.ForgotPasswordViewModel

/**
 * 🔑 ACTIVITY: ForgotPasswordActivity (Kotlin + Compose)
 * Pantalla para recuperar la contraseña del usuario.
 */
class ForgotPasswordActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        enableEdgeToEdge()
        super.onCreate(savedInstanceState)

        val prefilledEmail = intent.getStringExtra("email") ?: ""

        setContent {
            RutaGoTheme {
                val viewModel: ForgotPasswordViewModel = viewModel()
                val uiState by viewModel.uiState.collectAsState()

                // Prefijar correo si viene de Login
                LaunchedEffect(Unit) {
                    if (prefilledEmail.isNotEmpty()) {
                        viewModel.onEmailChanged(prefilledEmail)
                    }
                }

                ForgotPasswordScreen(
                    uiState = uiState,
                    onEmailChange = { viewModel.onEmailChanged(it) },
                    onRecoverClick = { viewModel.recoverPassword() },
                    onBackClick = { finish() },
                    onUnderstoodClick = { finish() }
                )
            }
        }
    }
}
